toolkit/tools/pkg/imagecustomizerlib/bootcustomizer.go

```diff
@@ -61,69 +101,107 @@ func (b *BootCustomizer) AddKernelCommandLine(extraCommandLine []string) error {
+	case bootConfigTypeGrubMkconfig:
 		defaultGrubFileContent, err := addExtraCommandLineToDefaultGrubFile(b.defaultGrubFileContent, combinedArgs)
 		if err != nil {
 			return err
 		}
 
 		b.defaultGrubFileContent = defaultGrubFileContent
-	} else {
+
+	case bootConfigTypeGrubLegacy:
 		// Add the args directly to the /boot/grub2/grub.cfg file.
 		grubCfgContent, err := appendKernelCommandLineArgsAll(b.grubCfgContent, combinedArgs)
 		if err != nil {
 			return err
 		}
 
 		b.grubCfgContent = grubCfgContent
+
+	case bootConfigTypeUki:
+		// UKI passthrough mode: preserve existing UKI boot configuration.
+		// Cmdline args are embedded in UKI files and managed through UKI regeneration,
+		// not through grub config files or boot loader configuration.
 	}
 
 	return nil
 }
 
-// Gets the image's configured SELinux mode.
-func (b *BootCustomizer) getSELinuxModeFromGrub() (imagecustomizerapi.SELinuxMode, error) {
+// getSELinuxModeFromCmdline gets the image's configured SELinux mode from kernel command-line.
+// Returns (mode, found, error) where:
+//   - mode: The detected SELinux mode (or SELinuxModeDefault if not found)
+//   - found: true if cmdline exists and was successfully parsed
+//   - error: Any error encountered during parsing
+func (b *BootCustomizer) getSELinuxModeFromCmdline(buildDir string, imageChroot safechroot.ChrootInterface) (imagecustomizerapi.SELinuxMode, bool, error) {
 	var err error
 	var args []grubConfigLinuxArg
 
 	// Get the SELinux kernel command-line args.
-	if b.isGrubMkconfig {
+	switch b.bootConfigType {
+	case bootConfigTypeGrubMkconfig:
 		// Check both GRUB_CMDLINE_LINUX and GRUB_CMDLINE_LINUX_DEFAULT variables
 		// and merge arguments from both if they exist
 		args, err = GetDefaultGrubFileLinuxArgsFromMultipleVars(b.defaultGrubFileContent)
 		if err != nil {
-			return "", fmt.Errorf("failed to find SELinux args in grub file (%s):\n%w", installutils.GrubDefFile, err)
+			return imagecustomizerapi.SELinuxModeDefault, false, fmt.Errorf("failed to find SELinux args in grub file (%s):\n%w", installutils.GrubDefFile, err)
 		}
-	} else {
+
+	case bootConfigTypeGrubLegacy:
 		args, _, err = getLinuxCommandLineArgs(b.grubCfgContent)
 		if err != nil {
-			return imagecustomizerapi.SELinuxModeDefault,
+			return imagecustomizerapi.SELinuxModeDefault, false,
 				fmt.Errorf("failed to parse SELinux args from grub file (%s):\n%w", installutils.GrubCfgFile, err)
 		}
+
+	case bootConfigTypeUki:
+		espDir := filepath.Join(imageChroot.RootDir(), EspDir)
+		tempDir := filepath.Join(buildDir, "uki-selinux-temp")
+
+		err := os.MkdirAll(tempDir, 0o755)
+		if err != nil {
+			return imagecustomizerapi.SELinuxModeDefault, false, fmt.Errorf("failed to create temp directory: %w", err)
+		}
+		defer os.RemoveAll(tempDir)
+
+		kernelToArgs, err := extractKernelCmdlineFromUkiEfis(espDir, tempDir)
+		if err != nil {
+			return imagecustomizerapi.SELinuxModeDefault, false, err
+		}
+
+		if len(kernelToArgs) == 0 {
```

Comment:
If we are in UKI passthrough mode and `kernelToArgs` is empty (no UKIs?), isn't something very wrong? Should we return an error here? (Or just let `parseFirstCmdlineFromUkiMap` return the error?)

Reply:
Yes, fixed, thanks!